src/test/java/com/spotify/futures/CompletableFuturesTest.java

```diff
@@ -51,12 +60,25 @@
 import static org.hamcrest.core.Is.isA;
 import static org.hamcrest.core.IsNot.not;
 import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 public class CompletableFuturesTest {
 
+  private DeterministicScheduler executor;
```

Comment:
This + using Mockito wouldn't be necessary if the code constructing the job for translating the polling operation into a `CompletableFuture` is decoupled from the scheduling the job.  Then the tests could be only about testing the functionality of that job, since you don't need to include testing that `ScheduledExecutorService.scheduleAtFixedRate` works.